components/centraldashboard-angular/frontend/public/library.js

```diff
@@ -113,7 +113,7 @@ class CentralDashboardEventHandler_ {
     }
 
     get isIframed() {
-        return this.window.location !== this.parent.location;
+        return this.window?.location !== this.parent?.location;
```

Comment:
After some digging I found that the reason `this.parent.location` is undefined is because Cypress changed it to `this.self.location`. If we change the [modifyObstructiveCode](https://docs.cypress.io/guides/references/configuration#modifyObstructiveCode) configuration to `false` then we can remove the optional chaining. @orfeas-k wdyt?

Reply:
Great find! Sounds like a much better solution than introducing optional chaining in a place we normally don't need just for the tests to pass. I 'll push some fix-ups with this included.